src/sql/src/pure.rs

```diff
@@ -1523,39 +1546,113 @@ async fn purify_create_table_from_source(
+                    scx.catalog.minimal_qualification(qualified_source_name),
+                    connection_name
+                )
+            };
 
-            let (_, desired_view_desc, output) = views.swap_remove(view_idx);
+            let mut views = load_gen_connection.load_generator.views();
+            if views.is_empty() {
+                // If there are no views then this load-generator just has a single output
+                // and the external reference specified in the statement should be the same
+                // as the load-generator's schema name or empty.
+                match external_reference {
+                    None => {}
+                    Some(reference) => {
+                        if reference.0.len() != 1
+                            || load_gen_connection.load_generator.schema_name()
+                                != reference.0.last().unwrap().as_str()
+                        {
+                            Err(LoadGeneratorSourcePurificationError::WrongLoadGenerator(
+                                load_gen_connection.load_generator.schema_name().to_string(),
+                                reference.clone(),
+                            ))?;
+                        }
+                    }
+                }
+                PurifiedSourceExport {
+                    external_reference: UnresolvedItemName(vec![Ident::new_unchecked(
+                        load_gen_connection.load_generator.schema_name(),
+                    )]),
+                    details: PurifiedExportDetails::LoadGenerator {
+                        table: None,
+                        output: LoadGeneratorOutput::Default,
+                    },
+                }
+            } else {
+                let external_reference = external_reference.as_ref().ok_or(
+                    LoadGeneratorSourcePurificationError::MultiOutputRequiresExternalReference,
+                )?;
+                // Resolve the desired view reference to a `schema_name.view_name` reference
+                let resolver = SourceReferenceResolver::new(
+                    LOAD_GENERATOR_DATABASE_NAME,
+                    &views
+                        .iter()
+                        .map(|(view_name, _, _)| {
+                            (load_gen_connection.load_generator.schema_name(), *view_name)
+                        })
+                        .collect_vec(),
+                )?;
+                let (qualified_reference, view_idx) = resolver.resolve(&external_reference.0, 2)?;
+
+                let (_, table, output) = views.swap_remove(view_idx);
+                PurifiedSourceExport {
+                    external_reference: qualified_reference,
+                    details: PurifiedExportDetails::LoadGenerator {
+                        table: Some(table),
+                        output,
+                    },
+                }
+            }
+        }
```

Comment:
This logic is the distinction between multi-output and single-output load-generators sources. Referencing my earlier comment in `plan/statement/ddl.rs`, multi-output load-generators behave similarly to mysql/postgres sources in that they determine their full output relations here during purification, whereas single-output load-generators behave like kafka sources. I think we could do more work in the future to try to unify these semantics.

This also implements the slight difference in requirements for the external-reference specification -- if it's a single-output load-generator then the reference is optional, and if it's not then it's required